Add AggregatedMetrics to support multiple Metrics implementations

Donnerbart opened the pull request:
Partially fixes #2884
Alternative to #2912

See https://github.com/operator-framework/java-operator-sdk/pull/2912#issuecomment-3228683758

Donnerbart commented on micrometer-support/src/test/java/io/javaoperatorsdk/operator/monitoring/micrometer/AggregatedMetricsTest.java:
```diff
@@ -0,0 +1,181 @@
+
+class AggregatedMetricsTest {
+
+  private final Metrics metrics1 = mock();
+  private final Metrics metrics2 = mock();
+  private final Metrics metrics3 = mock();
+  private final Controller<HasMetadata> controller = mock();
+  private final Event event = mock();
+  private final HasMetadata resource = mock();
+  private final RetryInfo retryInfo = mock();
+  private final ResourceID resourceID = mock();
+  private final Metrics.ControllerExecution<String> controllerExecution = mock();
```
I think this code style is much easier to read. Also to my knowledge it's one of the few approaches to define mocks for generic types without warnings or suppressions (maybe beside the `@Mock` annotation).

Donnerbart commented:
The new class should probably go into the same package as the `Metrics` interface. It has nothing to do with the concrete MicroMeter implementation. Will update the PR tomorrow.

metacosm commented on micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/AggregatedMetrics.java:
```diff
@@ -0,0 +1,106 @@
+ * All method calls are delegated to each metrics instance in the list in the order they were
+ * provided to the constructor.
+ *
+ * <p><strong>Important:</strong> The {@link #timeControllerExecution(ControllerExecution)} method
```
Good point. I forgot about that when I suggested this design! 😓

xstefank reviewed (approved):
LGTM

csviri commented:
Hi @Donnerbart loogks, great, would be better if you could target `next` branch, for the next minor version. Maybe also to mention this addon on docs. otherwise great, thank you

Donnerbart commented:
* Moved the class into the `io.javaoperatorsdk.operator.api.monitoring` package.
* Changed the target branch to `next`.
* Added and improved the documentation.
